### ISSUE TITLE
Every WriteMessages 1 second

### ISSUE THREAD

**Issue by teng231** [segmentio/kafka-go#1180]
**Describe the bug**

<img width="832" alt="image" src="https://github.com/segmentio/kafka-go/assets/20247951/cf3f2326-a921-4d69-b8c0-ec7f33d2070f">
see this code writer.go: 824
if i not set BatchTimeout duration of writeMessage is 1s

**Comment by petedannemann** [comment]
Yes, the default BatchTimeout for Writer is 1 second. Decreasing this can improve throughput. We can't change the default as that would be a breaking change
